mapswipe_workers/mapswipe_workers/firebase_to_postgres/transfer_results.py

```diff
@@ -340,6 +384,79 @@ def save_results_to_postgres(results_file, project_id, filter_mode: bool):
     logger.info("copied results into postgres.")
 
 
+def save_user_group_results_to_postgres(
+    user_group_results_file,
+    project_id,
+    filter_mode: bool,
+):
+    """
+    Saves results to a temporary table in postgres
+    using the COPY Statement of Postgres
+    for a more efficient import into the database.
+    Parameters
+    ----------
+    user_group_results_file: io.StringIO
+    filter_mode: boolean
+        If true, try to filter out invalid results.
+    """
+
+    p_con = auth.postgresDB()
+    columns = [
+        "project_id",
+        "group_id",
+        "user_id",
+        "user_group_id",
+    ]
+    user_group_results_file.seek(0)
+    p_con.copy_from(user_group_results_file, "results_user_groups_temp", columns)
+    user_group_results_file.close()
+
+    if filter_mode:
+        logger.warn(f"trying to remove invalid tasks from {project_id}.")
+
+        filter_query = """
+            WITH project_groups AS (
+                SELECT
+                    group_id
+                FROM groups
+                WHERE project_id = %(project_id)s
+            ),
+            -- Results for which we can't join a group from the groups table
+            -- are invalid. For these invalid results the group_id set by the app
+            -- is not correct. Hence, we delete these results from the
+            -- results_user_groups_temp table.
+            user_group_results_to_delete AS (
+                SELECT
+                    R.group_id,
+                    R.user_id
+                FROM results_user_groups_temp R
+                    LEFT join project_groups T on R.group_id = T.group_id
+                    LEFT join users U on R.user_id = U.user_id
+                    LEFT join user_groups UG on R.user_group_id = UG.user_group_id
+                WHERE T.group_id is NULL
+                    OR U.user_id is NULL
+                    OR UG.user_group_id is NULL
+            )
+            DELETE FROM results_user_groups_temp R1
+            USING user_group_results_to_delete R2
+            WHERE
+                R1.group_id = R2.group_id
+                AND R1.user_id = R2.user_id
+        """
+        p_con.query(filter_query, {"project_id": project_id})
+
+    query_insert_results = """
+        INSERT INTO results_user_groups
+            SELECT * FROM results_user_groups_temp
+        ON CONFLICT (project_id, group_id, user_id, user_group_id)
+        DO NOTHING;
+        -- TRUNCATE results_user_groups_temp; ??
```

Comment:
this is not needed, as we run this in a separate step.

Reply:
Sure. Will remove this.